docs/products/access/topics/oidc/claims-lifecycle/claims-lifecycle.md

```diff
@@ -0,0 +1,76 @@
+           profile is requested by another API call from the CIM, Access will pass the person identifier. Otherwise, the User object will be
+           persisted (see [#2.2.7.](#2.2.7.)).
+        5. The Person API response is mapped to a standard UserInfo format.
+        6. The User object is populated with any claims returned by the CIM Person API. If a given claim is already present in the User
+           object, it's overwritten with the one coming from the CIM Person Profile, which is considered more recent.
+        7. The User Object with all the claims is persisted.
+3. The authorization request is considered successful, Access generates the _authorization_code_.
+4. The OAuth Client / OIDC Relying Party receives a 302 Redirect with the _authorization_code_ from Access.
+
+## Token
+
+![AWS Configuration](img/access-claims-lifecycle-token.jpg)
+
+1. The OAuth Client / OIDC Relying Party requests the exchange of the _authorisation_code_ for an Access Token / ID Token.
+2. Access resolves a User Object that was persisted during the authorisation code flow.
+3. If an Access Token Customisation Web Hook is configured for a requesting OAuth Client/OIDC Relying Party, see [#3.1](#3.1.).
+    1. Retrieve user claim from the underlying Identity Provider (depending on the type of Identity Provider, different sources may be
+       considered here). If the user claims were fetched during the authorisation code flow, Access will resolve them from an internal
+       cache.
+    2. The User Object is populated with any claims returned by the User Info API. If a given claim already exists in the User Object, it
+       will be overwritten with the claim from the User Info source, which is considered more recent.
+    3. [Access Token Customization Web Hook](https://docs.onegini.com/products/access/topics/integration-extension/hooks/customize-access-token/customize-access-token-hook/)
+       is executed. The hook implementation is provided with all available user information, as well as contextual details about the Access
+       Token being created.
+4. If the underlying Identity Provider is of type OIDC and `openid` scope was requested, Access triggers ID Token assembly,
+   see [#4.1](#4.1.).
+    1. Access starts to assemble all the details needed to generate the ID token, using the internal User Object as the source of the user
+       information.
+    2. Retrieve user claims from the underlying Identity Provider if available, see [#4.3.](#4.3.), otherwise [#4.5.](#4.5.).
+    3. Retrieve user claims from the underlying Identity Provider (depending on the type of Identity Provider, different sources may be
+       considered here). If the user claims were fetched during the authorisation code flow, Access will resolve them from an internal
+       cache.
+    4. The User Object is populated with any claims returned by the User Info API. If a given claim already exists in the User Object, it is
```

Comment:
I'm not sure if we want to complicate this but `Map assertion attributes` for CIM is not turn on by default and this flow would look bit different when it's turned off.

Reply:
Hmm, I think it's worth including. I will have how to include it in the diagram 👍